ETC Ossification and Social Scalability (#1143)
ethereumclassic/ethereumclassic.github.io: content/blog/2023-03-29-ethereum-classic-ossification-and-social-scalability/index.md

---
title: "Ethereum Classic Ossification and Social Scalability"
date: 2023-03-29
author: Donald McIntyre
contributors: ["DonaldMcIntyre"]
tags: ["philosophy", "education"]
linkImage: ./etc-ossification-banner.png
---

---
**You can listen to or watch this video here:**

<iframe width="560" height="315" src="https://www.youtube.com/embed/wfas7sUqnzs" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen></iframe>

---

The whole purpose of blockchains is decentralization. 

Decentralization means to reduce the dependency on any particular trusted third party to act as intermediary in economic transactions between individuals and businesses.

The reason for this trust minimization is to reduce the possible restrictions that trusted third parties usually end up imposing on users of the system. 

> Restrictions that trusted third parties usually impose may range from transactional costs, lock-in costs, to even blocking access to services altogether.

Author: It is "lock-in costs". It is a term in tech that people are locked into apps and can't leave because the cost of moving is too high. I will not merge this commit because we need to publish tomorrow, but we can make the changes after pulishing.

In the case of networks as Ethereum Classic, the attack surface problem has three angles: Complexity, politics, and change.

## Complexity

The complexity attack surface: The more complex rules and code needed to build the system, the more hidden ways hackers will find to attack it.

Complexity also raises the risk of hidden bugs or unforeseen failures.

The way of reducing this surface is by reducing complexity or reaching a point of ossification where the system is functional enough to support all new features and innovation in external complementary protocols.

Examples of this may be rollups, channels, and other scaling systems built on top of ETC, instead of sharding it internally or integrating them into the network.

## Protocol Politics

The argument attack surface: The more decisions to argue over, the more the opportunity to convince the ecosystem to accept bad upgrades that reduce decentralization.

> In blockchain networks there are several constituents that participate in the debates when there are new proposals for change. These debates may become very passionate and even toxic at times. Sometimes, parties with enough budgets, charisma, or influence may actually convince participants to make horrible changes. The most famous of these examples is the migration of Ethereum from proof of work to proof of stake, a totally prone to centralization system.

Reviewer: Line 79: Add hyphens; Rewrite for clarity
```suggestion
In blockchain networks there are several constituents that participate in the debates when there are new proposals for change. These debates may become very passionate and even toxic at times. Sometimes, parties with enough budgets, charisma, or influence may actually convince participants to make horrible changes. The most famous of these examples is the migration of Ethereum from proof-of-work to proof-of-stake, a system prone to total centralization.
```

These processes turn developers, economic node operators, and miners into political animals fighting for their self interest.

Ossification would avoid this problem entirely as there would be no possibility of tricking the ecosystem with fancy arguments and gimmicks anymore.

## Change

The change attack surface: The more changes and upgrades there are, the more opportunities to introduce faulty rules or bugs.

> Similar to protocol politics, just the frequency of change, even if necessary and benevolent, is a risk to the system. 

Reviewer: Line 89: Reword awkward stucture
```suggestion
Similar to protocol politics, the frequency of changes made, even if necessary and benevolent, is a risk to the system. 
```

> It has happened in the past, for example, that Bitcoin had a bug nobody had seen, and when it was discovered they had to introduce an urgent change.

Reviewer: Line 91: Reword for clarity
```suggestion
As an example, Bitcoin once had a critical bug introduced as part of an update. When it was discovered,  deployment of an urgent hot-fix was required to patch the error.
```

> Save hidden bugs that may still be there, ossification would eliminate this risk, at least of introducing new bugs and errors, because there would be no introduction of new code or rules to the protocol.

Reviewer: Line 93: Reword awkward structure
```suggestion
Except for those hidden and already present in the code, ossification eliminates the introduction of new bugs or errors because there would be no new code or rules deployed to the protocol.
```

## Striking a Balance

The best way to strike a balance between trust minimization, change, complexity, and protocol politics is to reduce change by reaching protocol ossification and letting all new features and design goals to be built on higher layers.

> This is how we achieve the most social scalability.

Reviewer: Line 99: Reword for clarity
```suggestion
In this way, the highest level of social scalability is achieved.
```
